### PR TITLE
Final Version of Capstone Project

### PR DESCRIPTION
**Pull request by Sahar-SE**
### This PR Contains:

- The Weather Condition API.
- Build files structure with React app.
- Routes and navigation for app
- Data come from the API.
- The tests for the application.
- Style components to match the design provided.
- Deploy the project and test for final details.
- Record a video of project.
- Create a good README.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+- JavaScript
+- React
+- react-redux
+- Webpack
+- Jest
+
+## Live Demo (Heroku) 💻
+- [Live Demo](https://redux-h.herokuapp.com/)
+- [Video](https://drive.google.com/file/d/1Tzbxlkd5XIjPWfbDMQomM3nNWWKI_xxj/view?usp=sharing)
+
```
Comment (RolandM99) [inline README.md]: - [x] Great Job by adding a video presentation of your Project. To make it again more interesting I'll suggest you add in your presentation video a highlight about a piece of the code you find interesting or you're proud of making.

**Review by RolandM99 — CHANGES_REQUESTED** [review]
Hello @Sahar-SE, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

### To highlight 💯 
- Good commit message ✔️ 
- No Linter errors ✔️ 
- Follow Gitflow ✔️ 

## Required Changes ♻️
_Check the comments under the review._

- [x] [REQUIRED] Kindly, add Unit tests for `States` function in the StatesPage directory from your components. Also, add Unit tests for some of your pure functions in your Redux code.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me on @Rolandm99 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by roshan-bajgain — CHANGES_REQUESTED** [review]
Hi @Sahar-SE, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
- No linter errors. ✔️
- GitHub flow is followed. ✔️
- Professional README file. ✔️
- PR has a good title and summary. ✔️

 ## Required Changes ♻️

- [x] You did great work and some of the tests are passing as well but you don't have any test on States.test.js and It is throwing errors kindly write the test and ask for another review
### Error
ss



## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by alexander16108 — CHANGES_REQUESTED** [review]
Hi @Sahar-SE , 

Great work on making the changes 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.



## Suggested changes
_Check the comments under the review._

 -  As per the requirement, you should test the pure functions in redux code. by performing a mock data fetching and Redux store connection. 👍🏾 


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Pull request merged**